clarity/contracts/vaults-v2/arkadiko-vaults-sorted-v1-1.clar

```diff
@@ -0,0 +1,279 @@
+    (is-eq caller (contract-call? .arkadiko-dao get-dao-owner))
+  )
+)
+
+;; ---------------------------------------------------------
+;; Find position
+;; ---------------------------------------------------------
+
+;; TODO: check the cost
+;; Find the actual position given prev/next hints.
+;; Hints are kept off chain. But the list can change within the same block.
+;; So we use the prev/next hints to find the actual position. 
+(define-read-only (find-position (owner principal) (token principal) (nicr uint) (prev-owner-hint (optional principal)) (next-owner-hint (optional principal)))
+  (let (
+    (prev-owner-1-temp (if (is-some prev-owner-hint) (get prev-owner (get-vault (unwrap-panic prev-owner-hint) token)) none))
+    (prev-owner-1 (if (is-some prev-owner-1-temp) (unwrap-panic prev-owner-1-temp) none))
+    (prev-owner-2-temp (if (is-some prev-owner-1) (get prev-owner (get-vault (unwrap-panic prev-owner-1) token)) none))
+    (prev-owner-2 (if (is-some prev-owner-2-temp) (unwrap-panic prev-owner-2-temp) none))
+    (prev-owner-3-temp (if (is-some prev-owner-2) (get prev-owner (get-vault (unwrap-panic prev-owner-2) token)) none))
+    (prev-owner-3 (if (is-some prev-owner-3-temp) (unwrap-panic prev-owner-3-temp) none))
+    (prev-owner-4-temp (if (is-some prev-owner-3) (get prev-owner (get-vault (unwrap-panic prev-owner-3) token)) none))
+    (prev-owner-4 (if (is-some prev-owner-4-temp) (unwrap-panic prev-owner-4-temp) none))
+    (prev-owner-5-temp (if (is-some prev-owner-4) (get prev-owner (get-vault (unwrap-panic prev-owner-4) token)) none))
+    (prev-owner-5 (if (is-some prev-owner-5-temp) (unwrap-panic prev-owner-5-temp) none))
+
+    (next-owner-1-temp (if (is-some next-owner-hint) (get next-owner (get-vault (unwrap-panic next-owner-hint) token)) none))
+    (next-owner-1 (if (is-some next-owner-1-temp) (unwrap-panic next-owner-1-temp) none))
+    (next-owner-2-temp (if (is-some next-owner-1) (get next-owner (get-vault (unwrap-panic next-owner-1) token)) none))
+    (next-owner-2 (if (is-some next-owner-2-temp) (unwrap-panic next-owner-2-temp) none))
+    (next-owner-3-temp (if (is-some next-owner-1) (get next-owner (get-vault (unwrap-panic next-owner-2) token)) none))
+    (next-owner-3 (if (is-some next-owner-3-temp) (unwrap-panic next-owner-3-temp) none))
+    (next-owner-4-temp (if (is-some next-owner-1) (get next-owner (get-vault (unwrap-panic next-owner-3) token)) none))
+    (next-owner-4 (if (is-some next-owner-4-temp) (unwrap-panic next-owner-4-temp) none))
+    (next-owner-5-temp (if (is-some next-owner-1) (get next-owner (get-vault (unwrap-panic next-owner-4) token)) none))
+    (next-owner-5 (if (is-some next-owner-5-temp) (unwrap-panic next-owner-5-temp) none))
+  )
+    ;; First check if given prev/next is correct
+    (if (get correct (check-position owner token nicr prev-owner-hint next-owner-hint)) { prev: prev-owner-hint, next: next-owner-hint }
```

Comment:
Q: I'm not sure how clarity executes this function (statement order) but we don't need to query all the `next-owner-temp` logic if this turns out to be correct already? I think we need nested `let` statements? Not sure if that lowers cost but probably will in some cases?

Reply:
A: fixed